pkg/lib/saml/binding/flate.go

```diff
@@ -0,0 +1,33 @@
+// Copied from https://github.com/crewjam/saml/blob/8e9236867d176ad6338c870a84e2039aef8a5021/flate.go
```

Comment:
I suggest we organize things as follows

- pkg/lib/saml/samlprotocol: This package defines the structs that we use. It can be imported by others. Error, Request, Response, constants, etc, all live here.
- pkg/lib/saml/samlbinding: This package handles HTTPRedirect and HTTPPost binding. It can import pkg/lib/saml/samlprotocol
- pkg/lib/saml: This package contains the SAML service struct. It can import pkg/lib/saml/samlprotocol and pkg/lib/saml/samlbinding